framework/src/media/MediaPlayerImpl.cpp

```diff
@@ -541,16 +548,30 @@ void MediaPlayerImpl::notifyObserver(player_observer_command_t cmd)
 
 void MediaPlayerImpl::playback()
 {
-	ssize_t num_read = mInputDataSource->read(mBuffer,(int)mBufSize);
+	ssize_t num_read = mInputDataSource->read(mBuffer, (int)mBufSize);
 	meddbg("num_read : %d\n", num_read);
 	if (num_read > 0) {
-		start_audio_stream_out(mBuffer, get_output_bytes_to_frame((unsigned int)num_read));
+		int ret = start_audio_stream_out(mBuffer, get_output_bytes_to_frame((unsigned int)num_read));
+		if (ret < 0) {
+			PlayerWorker &mpw = PlayerWorker::getWorker();
+			switch (ret) {
+			case AUDIO_MANAGER_XRUN_STATE:
```

Comment:
This is very important error value, to inform this to user, we had better add one more parameter to send AUDIO_MANAGER_XRUN_STATE message.
or we can use last boolean param .
```
mpw.enQueue(&MediaPlayerImpl::stopPlayer, shared_from_this(), RECORDER_ERROR /* RECORDER_ERROR_XRUN later */); 
...

void MediaPlayerImpl::stopPlayer(recorder_result_t result) {
}
```

Reply:
Now I just send `AUDIO_MANAGER_XRUN_STATE`.